From eb422ab5e07498a7a8d086f6a942ee35ab3c9776 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Thu, 9 Dec 2021 17:35:24 +0100
Subject: [PATCH 34/35] Fix backport, context destruction was omitted

When cherry-picking for 9df11e79b46c77d8c83f765b2a8e85b639fd55a2, this
line got removed when rebasing the patch, which created a leak.

BUG: 442844
Change-Id: Id00e8b194dcd910c5f01ce9d2cc318f96a4129a2
---
 src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp b/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
index 95d1049c..683fe123 100644
--- a/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
+++ b/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
@@ -407,6 +407,7 @@ QWaylandGLContext::~QWaylandGLContext()
 {
     delete m_blitter;
     m_blitter = nullptr;
+    eglDestroyContext(m_eglDisplay, m_context);
     if (m_decorationsContext != EGL_NO_CONTEXT)
         eglDestroyContext(m_eglDisplay, m_decorationsContext);
 }
-- 
2.34.1

